### PR TITLE
[WIP] Add Dockerfile and deployment files

### PR DESCRIPTION
**Pull request by prateekpandey14**
- Dockerfile to build docker image
- Kubernetes deploy yaml files

Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

### CODE REVIEW

**Comment by msau42** [comment]
Can you also modify the Makefile according to the comments [here](https://github.com/kubernetes-csi/csi-driver-iscsi/blob/master/Makefile#L17)?

**Comment by prateekpandey14** [comment]
@msau42 updated the Makefile with required changes...thanks

File: deploy/kubernetes/csi-iscsi-attacher.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          env:
+            - name: ADDRESS
+              value: /csi/csi.sock
+          imagePullPolicy: "IfNotPresent"
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /csi
+
+        - name: iscsi
+          image: quay.io/k8scsi/iscsiplugin:v1.0.0
```
Comment (prateekpandey14) [inline deploy/kubernetes/csi-iscsi-attacher.yaml]: @msau42 should be v0.1.0 ?

Comment (lyyao09) [reply deploy/kubernetes/csi-iscsi-attacher.yaml]: > @msau42 should be v0.1.0 ?
the tag is wrong~

**Pull request closed without merging**